sys/build.rs

```diff
@@ -55,58 +61,79 @@ fn main() {
 
     // Download
     let src = download_and_unzip(&version_dlib);
+    build_dlib(&src);
+}
 
-    // Build
-    let dst = cmake::Config::new(src.join("examples"))
-        .no_build_target(true)
-        .define("DLIB_JPEG_SUPPORT", "1")
-        .define("DLIB_PNG_SUPPORT", "1")
-        // .define("DLIB_USE_BLAS", "1")
-        // .define("DLIB_USE_LAPACK", "1")
-        .define("USE_AVX_INSTRUCTIONS", "1")
-        .define("USE_SSE2_INSTRUCTIONS", "1")
-        .define("USE_SSE4_INSTRUCTIONS", "1")
+fn build_dlib(src: &PathBuf) {
+    let dst = cmake::Config::new(&src)
+        .no_build_target(false)
+        .define("JPEG_INCLUDE_DIR", src.join("dlib").join("external").join("libjpeg"))
+        .define("JPEG_LIBRARY", src.join("dlib").join("external").join("libjpeg"))
+        .define("PNG_PNG_INCLUDE_DIR", src.join("dlib").join("external").join("libpng"))
```

Comment:
libpng: The linux repositories (such as Ubuntu) without X11 will be broken.

Reply:
Yes, I ll take a look.